### PR TITLE
[misc][frontend] log all available endpoints

### LINKED ISSUE #6194: [Bug]: No end point available after model is fully loaded
### Your current environment

```text
vllm==0.5.0.post1
vllm-flash-attn==2.5.9
Python 3.10.11
```

### 🐛 Describe the bug

Hey Guys,
I loaded a Llama-3-70B with vllm, once the model is loaded, I get the following message, which I assume means it is ready to take the requests,
```
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
```
However, no end point is available, I tried both / and /generate, both giving me 404 error, here is the log printed by vllm:
```
INFO:     127.0.0.1:55504 - "GET / HTTP/1.1" 404 Not Found
INFO 07-07 20:12:44 metrics.py:295] Avg prompt throughput: 0.0 tokens/s, Avg generation throughput: 0.0 tokens/s, Running: 0 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
INFO:     127.0.0.1:59820 - "GET /generate HTTP/1.1" 404 Not Found
INFO 07-07 20:12:54 metrics.py:295] Avg prompt throughput: 0.0 tokens/s, Avg generation throughput: 0.0 tokens/s, Running: 0 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
INFO:     127.0.0.1:53798 - "POST /generate HTTP/1.1" 404 Not Found
```

### PR DESCRIPTION
**Pull request by youkaichao**
fixes https://github.com/vllm-project/vllm/issues/6194

quite a lot users don't know the difference between simple api server and openai api server. I got several issue reports recently, that users are manually getting `/generate` from openai api server, or getting `/v1/models` from simple api server, and then complain about 404 response.

logging all available endpoints should help users clearly know what endpoints they can test.

example output:

simple api server: `python -m vllm.entrypoints.api_server`

```text
INFO 07-07 13:50:36 api_server.py:116] Available endpoints are:
INFO 07-07 13:50:36 api_server.py:120] Path: /openapi.json, Methods: GET, HEAD
INFO 07-07 13:50:36 api_server.py:120] Path: /docs, Methods: GET, HEAD
INFO 07-07 13:50:36 api_server.py:120] Path: /docs/oauth2-redirect, Methods: GET, HEAD
INFO 07-07 13:50:36 api_server.py:120] Path: /redoc, Methods: GET, HEAD
INFO 07-07 13:50:36 api_server.py:120] Path: /health, Methods: GET
INFO 07-07 13:50:36 api_server.py:120] Path: /generate, Methods: POST
```

openai api server: `python -m vllm.entrypoints.openai.api_server`

```text
INFO 07-07 13:53:48 api_server.py:244] Available endpoints are:
INFO 07-07 13:53:48 api_server.py:248] Path: /openapi.json, Methods: HEAD, GET
INFO 07-07 13:53:48 api_server.py:248] Path: /docs, Methods: HEAD, GET
INFO 07-07 13:53:48 api_server.py:248] Path: /docs/oauth2-redirect, Methods: HEAD, GET
INFO 07-07 13:53:48 api_server.py:248] Path: /redoc, Methods: HEAD, GET
INFO 07-07 13:53:48 api_server.py:248] Path: /metrics, Methods: No methods
INFO 07-07 13:53:48 api_server.py:248] Path: /health, Methods: GET
INFO 07-07 13:53:48 api_server.py:248] Path: /tokenize, Methods: POST
INFO 07-07 13:53:48 api_server.py:248] Path: /detokenize, Methods: POST
INFO 07-07 13:53:48 api_server.py:248] Path: /v1/models, Methods: GET
INFO 07-07 13:53:48 api_server.py:248] Path: /version, Methods: GET
INFO 07-07 13:53:48 api_server.py:248] Path: /v1/chat/completions, Methods: POST
INFO 07-07 13:53:48 api_server.py:248] Path: /v1/completions, Methods: POST
INFO 07-07 13:53:48 api_server.py:248] Path: /v1/embeddings, Methods: POST
```

### CODE REVIEW

File: vllm/entrypoints/api_server.py [hunk 1]
```diff
@@ -107,6 +110,13 @@ async def stream_results() -> AsyncGenerator[bytes, None]:
         engine_args, usage_context=UsageContext.API_SERVER)
 
     app.root_path = args.root_path
+
+    logger.info("Available endpoints are:")
+    for route in app.routes:
+        methods = ', '.join(route.methods) if hasattr(
+            route, 'methods') else 'No methods'
```
Comment (comaniac) [inline vllm/entrypoints/api_server.py:117]: What routes would have no method?

File: vllm/entrypoints/api_server.py [hunk 2]
```diff
@@ -107,6 +110,13 @@ async def stream_results() -> AsyncGenerator[bytes, None]:
         engine_args, usage_context=UsageContext.API_SERVER)
 
     app.root_path = args.root_path
+
+    logger.info("Available endpoints are:")
```
Comment (comaniac) [inline vllm/entrypoints/api_server.py:114]: Maybe more align to the terms we are using.
```suggestion
    logger.info("Available routes are:")
```

File: vllm/entrypoints/api_server.py (hunk 1 above)
Comment (youkaichao) [reply vllm/entrypoints/api_server.py]: fixed in [b64797c](https://github.com/vllm-project/vllm/pull/6195/commits/b64797cbb5d45dc2588dc1071b0b4d63dd830350). maybe this is for edge cases? chatgpt adds it, and i think it is fine to add the handling logic :)

Comment (comaniac) [reply vllm/entrypoints/api_server.py]: I'm mostly just curious lol
Then maybe we could just use "N/A" for concise.

Comment (comaniac) [reply vllm/entrypoints/api_server.py]: p.s. Filtering out is also fine.

**Review by comaniac — APPROVED** [review]
LGTM

**Comment by youkaichao** [comment]
merge as failed tests are caused by hf connection error.

**Pull request merged**